interfaces/builtin/account_control.go

```diff
@@ -56,26 +65,59 @@ capability fsetid,
 /var/log/faillog rwk,
 `
 
-// Needed because useradd uses a netlink socket
-const accountControlConnectedPlugSecComp = `
-# useradd requires chowning to 'shadow'
-fchown - u:root g:shadow
-fchown32 - u:root g:shadow
+// Needed because useradd uses a netlink socket, {{group}} is used as a
+// placeholder argument for the actual ID of a group owning /etc/shadow
+const accountControlConnectedPlugSecCompTemplate = `
+# useradd requires chowning to 0:'{{group}}'
+fchown - u:root {{group}}
+fchown32 - u:root {{group}}
 
 # from libaudit1
 bind
 socket AF_NETLINK - NETLINK_AUDIT
 `
 
+type accountControlInterface struct {
+	commonInterface
+	secCompSnippet string
+}
+
+func makeAccountControlSecCompSnippet() (string, error) {
+	gid, err := osutil.FindGidOwning("/etc/shadow")
+	if err != nil {
+		return "", err
+	}
+
+	snippet := strings.Replace(accountControlConnectedPlugSecCompTemplate,
+		"{{group}}", strconv.FormatUint(gid, 10), -1)
+
+	return snippet, nil
+}
+
+func (iface *accountControlInterface) SecCompConnectedPlug(spec *seccomp.Specification, plug *interfaces.Plug, Attrs map[string]interface{}, slot *interfaces.Slot, slotAttrs map[string]interface{}) error {
+	if iface.secCompSnippet == "" {
+		snippet, err := makeAccountControlSecCompSnippet()
+		if err != nil {
+			return err
+		}
+		iface.secCompSnippet = snippet
+	}
+	spec.AddSnippet(iface.secCompSnippet)
+	return nil
+}
+
 func init() {
-	registerIface(&commonInterface{
-		name:                  "account-control",
-		summary:               accountControlSummary,
-		implicitOnCore:        true,
-		implicitOnClassic:     true,
-		baseDeclarationSlots:  accountControlBaseDeclarationSlots,
-		connectedPlugAppArmor: accountControlConnectedPlugAppArmor,
-		connectedPlugSecComp:  accountControlConnectedPlugSecComp,
-		reservedForOS:         true,
+	registerIface(&accountControlInterface{
+		commonInterface: commonInterface{
```

Comment:
If you put this on the preceding line does `go fmt` let it through? If so this could be a nice way to address common/custom interface transitions without big deltas.